src/runtime/tests/manifest-parser-test.ts

```diff
@@ -875,6 +887,20 @@ describe('manifest parser', () => {
         bar: writes Bar {y: Number} = from p in foo.x select new Bar {y: p.num / 2}
       `);
     });
+    it('allows function calls as qualifiers', () => {
+      parse(`
+      particle Converter
+        foo: reads Foo {x: Number}
+        bar: writes Bar {y: Number} = from p in foo.x select new Bar {y: first(foo).x}
```

Comment:
It seems to me that we should modify ExpressionEntityField to accept a PaxelExpressionWithRefinement, so that for a schema `Thing {nums: [Number]}` you can write for example: `new Thing {nums: (from a in foo where a.x > 5 select a.x)}`. WDYT?

Reply:
Yeah we could do that, Kotlin version already allows this.

Reply:
Done